testruby.c

```diff
@@ -1,24 +1,51 @@
 #include <stdio.h>
 #include <stdlib.h>
-int i, M, N, x, y, j, z, var, red, st1, st2, st, red1, red2, l, d;
-char c = 'A', b, pamti, boja;
-char **tablica;
-void regija(int x, int y, char boja, char pamti)
+#include <stdbool.h>
+int M, N;
+char **matrix;
+bool isPixelValid(int x, int y)
 {
-    if (tablica[x][y] == boja)
+    if (x < 0 || x > N - 1 || y < 0 || y > M - 1)
+        return false;
+    else
+        return true;
+}
+char getPixelColor(int x, int y)
+{
+    if (isPixelValid(x, y) == true)
+        return matrix[x][y];
+        else return '\0';
+}
+char setPixelColor(int x, int y, char color)
+{
+    if (isPixelValid(x, y) == true)
+    {
+        matrix[x][y] = color;
+        return color;
+    }
+    else
+        return '\0';
+}
+void region(int x, int y, char b, char remember)

```

Comment:
What is the meaning of 'region'? It is a good practice to have descriptive names of the methods/functions in this case much better fit for a name would be fillRegion(...)

Reply:
are we sure 'b' is the best fit for the argument name?